agent/command/s3_copy.go

```diff
@@ -75,105 +95,140 @@ type s3Loc struct {
 	Path string `mapstructure:"path" plugin:"expand"`
 }
 
-func s3CopyFactory() Command   { return &s3copy{} }
-func (c *s3copy) Name() string { return "s3Copy.copy" }
+func s3CopyFactory() Command      { return &s3copy{} }
+func (s3pc *s3copy) Name() string { return "s3Copy.copy" }
 
-// ParseParams decodes the S3 push command parameters that are
-// specified as part of an S3CopyPlugin command; this is required
-// to satisfy the 'Command' interface
-func (c *s3copy) ParseParams(params map[string]interface{}) error {
-	if err := mapstructure.Decode(params, c); err != nil {
-		return errors.Wrapf(err, "error decoding %v params", c.Name())
+// ParseParams decodes the S3 push command parameters
+func (s3pc *s3copy) ParseParams(params map[string]interface{}) error {
+	decoder, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{
```

Comment:
What was the reason for using this method of decoding?

Reply:
There was no specific reason, I copied it from the s3put command. Would you prefer a different method?

Reply:
I was just wondering, since it's different than the existing method of decoding, but it's probably fine then.